### PR TITLE
feat: Create shipping options workflow

### PR DESCRIPTION
**Pull request by adrien2p**
**What**

- Create main workflow
- Update create pricing rule types step to be idempotent
- update remote joiner to support granular isList confguration on field aliases
- Add full workflow integration tests

### CODE REVIEW

**Comment by adrien2p** [comment]
@olivermrbl what do you think if we get this PR merged and I ll do the end point and tests in a separate pr just to not make this one bigger?

**Comment by olivermrbl** [comment]
@adrien2p , I think that makes sense

File: integration-tests/modules/__tests__/fulfillment/workflows/create-shipping-options.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        expect(prices).toContainEqual(
+          expect.objectContaining({
+            currency_code: "eur",
+            amount: 100,
+            rules_count: 1,
+          })
+        )
+      })
+
+      it.only("should revert the shipping options and prices", async () => {
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/fulfillment/workflows/create-shipping-options.ts]:
```suggestion
      it("should revert the shipping options and prices", async () => {
```

**Review by olivermrbl — APPROVED** [review]
Very nice, LGTM!

**Pull request merged**